File: scala/src/main/scala/ExcelSheetColumnNumber.scala
// Problem ID: 171
// URL: https://leetcode.com/problems/excel-sheet-column-number/

package leetcode

object ExcelSheetColumnNumber {
    def solution(columnTitle: String): Int = columnTitle.foldLeft(0)((acc, cur) => acc * 26 + (cur - 'A' + 1))
}

File: javascript/src/ExcelSheetColumnNumber.js
// Problem ID: 171
// URL: https://leetcode.com/problems/excel-sheet-column-number/

const solution = (columnTitle) => columnTitle.split('').reduce((acc, cur) => acc * 26 + (cur.charCodeAt(0) - 'A'.charCodeAt(0) + 1), 0)

module.exports = {
    solution,
};
